packages/core/src/zip/select.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+import R from 'ramda'
+
+import * as zip from './impl'
+
+import { childAt } from './skele/motion'
+
+import { isOfKind } from '../data'
+
+export const isStringArray = R.allPass([
+  R.is(Array),
+  R.or(R.all(R.is(String)), R.propEq('length', 0)),
+])
+
+const isTrue = R.allPass([R.is(Boolean), R.equals(true)])
+const isLocation = loc =>
+  loc && loc.meta && loc.meta.isBranch && loc.meta.children && loc.meta.makeNode
+export const isLocationArray = R.allPass([
+  R.is(Array),
+  R.complement(R.isEmpty),
+  R.all(isLocation),
+])
+
+// Predicate = String | [String] | Function :: Location -> Boolean | Location | [Location]
+// select :: [Predicate] -> Location -> [Location]
+export const select = (...predicates) => location => {
+  let result = [location]
```

Comment:
Could be 
```
let result = Seq.of(location)
```

Reply:
Also, not too keen on 

```
export const select = (...predicates) => location => { ... }
```

As an API signature. I get we can't have a curried fn with varargs but please decide if:

```
select([preds], loc)  /// [curried] 
```

or

```
select(...preds, loc) // [uncurried]
```

is your food of choice.

Reply:
`Rest parameter must be last formal parameter`

hence will go with:
```
select([preds], loc)  // [curried] 
```